core/model.py

```diff
@@ -0,0 +1,230 @@
+
+class Model():
+    _id: ObjectId
+    created_at: datetime
+    updated_at: datetime
+
+    def __init__(self, data: dict=dict()):
+        database = get_database()
+        self._database = database()
+        self._fields = self._get_fields()
+        self._set_value_from_data(data)
+
+    def _set_value_from_data(self, data: dict):
+        for key in self._fields:
+            setattr(self, key, data.get(key))
+
+    def _get_data_load_object(self, data: dict) -> dict:
+        data_load_object = {}
+        if not data:
+            return data_load_object
+        for key in data:
+            if key in _filter_out_load_object_key or key not in data or key not in self._fields:
+                continue
+
+            value = data[key]
+            if key in _prefilled_fields: # is _id
+                data_load_object[key] = value
+            elif self._is_reference_field(key): # is one2many reference
+                elements = []
+                reference_model = self._get_reference_model_of_field(key)
+                for element in value:
+                    elements.append(reference_model.get(element))
+                data_load_object[key] = elements
+            elif self.__annotations__[key].__base__ is Model: # is many2one reference
+                data_load_object[key] = self.__annotations__[key].get(value)
+            elif self.__annotations__[key].__base__ is Enum: # is enum
+                data_load_object[key] = self.__annotations__[key](value)
+            else: # is general type
+                data_load_object[key] = value
+        return data_load_object
+
+    def _get_data_store_object(self) -> dict:
+        data_store_object = {}
+        for key in self.__dict__.keys():
+            if key in _filter_out_store_object_key:
+                continue
+
+            value = self.__dict__[key]
+            if value is None:
+                    continue
+            if self._is_reference_field(key): # is one2many reference
+                element_ids = []
+                for element in value:
+                    if not element._is_saved():
+                        element.save()
+                    element_ids.append(element._id)
+                data_store_object[key] = element_ids
+            elif isinstance(value, Model): # is many2one reference
+                if not value._is_saved():
+                    value.save()
+                data_store_object[key] = value._id
+            elif isinstance(value, Enum): # is enum
+                data_store_object[key] = value.value
+            else: # is general type
+                data_store_object[key] = value
+        return data_store_object
+
+    def _is_saved(self) -> bool:
+        return hasattr(self, '_id') and self._id
+    
+    def _is_reference_field(self, field: str) -> bool:
+        return field in self.__annotations__ and len(get_args(self.__annotations__[field])) == 1 and isinstance(get_args(self.__annotations__[field])[0](), Model)
```

Comment:
1. What exactly does `len(get_args(self.__annotations__[field])) == 1` do?
2. Can't `isinstance(get_args(self.__annotations__[field])[0](), Model)` be replaced with `issubclass(...)`? This creates an object just to check it's type which is a very bad approach.

Reply:
This field is generic type, so user specify the type as `list[User]`

1. This check for length of `get_args`, if it is generic type it will return length 1, otherwise 0
    i.e. `list[User]` => 1, `str` => 0
   getting some value in empty list will raise error.
2. Yep. Changed to `issubclass`

0d30a47323334a25a1984fdcf6e4ff261288b8f7